server/src/main/java/org/elasticsearch/index/mapper/CompletionFieldMapper.java

```diff
@@ -369,8 +372,20 @@ public CompletionFieldType fieldType() {
         return (CompletionFieldType) super.fieldType();
     }
 
-    static PostingsFormat postingsFormat() {
-        return PostingsFormat.forName("Completion99");
+    public static PostingsFormat postingsFormat() {
+        return PostingsFormatHolder.POSTINGS_FORMAT;
+    }
+
+    private static class PostingsFormatHolder {
+        private static final PostingsFormat POSTINGS_FORMAT = getPostingsFormat();
+
+        private static PostingsFormat getPostingsFormat() {
+            String defaultName = "Completion99";
+            String codecName = ExtensionLoader.loadSingleton(ServiceLoader.load(PostingsFormatExtension.class))
+                .map(PostingsFormatExtension::getCompletionPostingsFormatName)
+                .orElse(defaultName);
+            return PostingsFormat.forName(codecName);
+        }
```

Comment:
Previously, the `FieldMapper` cached the results of the `postingsFormat()` function, which is called multiple times when flushing/merging (IIUC).  Now that this codepath has been removed, I've retained this caching behaviour  via this `PostingsFormatHolder`.